_targets/build.common

```diff
@@ -133,9 +147,17 @@ b_add2img() {
 	: "${OFFSET:?variable unset}"    # starting file's position in the image
 	: "${SIZE_PAGE:?variable unset}"
 	: "${PAGE_MASK:?variable unset}"
+	: "${FLASH_IMAGE:?variable unset}"
 
 	printf "Copying %s (offs=%dB)\n" "$1" "$OFFSET"
-	dd if="$1" of="$2" bs=1 seek="$OFFSET" conv=notrunc 2>/dev/null
+	if [ "$FLASH_IMAGE" = 'y' ]; then
+		b_pad_ff "$2" "$OFFSET"
+		touch "$2"
```

Comment:
- why touch is needed?
- the file copying could be also done by the same `dd` (outside of if-else) - that would highlight that the `FLASH_IMAGE` param changes only padding value
- I would stick to one `printf` for consistency - regardless of the input params. The `printf` could obviously be extended, I have eg. `printf "Copying %-100s (offs=0x%08x) to %s\n" "$1" "$OFFSET" "$(basename "$2")"` in stash. Probably info about added padding size could also be beneficial)

Reply:
1. you can safely redirect to nonexistent file, shell will open(O_CREAT) it for you (eg. `cat /etc/passwd >> /tmp/nonexistent` works fine
2. I think `bs=1` is not needed here (dd will make the last block smaller without an issue) - using `cat` + redirect is AFAIK similar to using `bs=4096` (my personal preference would be to use `dd` - that way it's possible to add something in the middle /at the beginning of the image - now you can only append at the end)